mpmath/functions/bessel.py

```diff
@@ -82,6 +82,8 @@ def h(n):
 def besseli(ctx, n, z, derivative=0, **kwargs):
     n = ctx.convert(n)
     z = ctx.convert(z)
+    if ctx.nint(n) and n < 0:
+        return ctx.besseli(-n, z, derivative, **kwargs)
```

Comment:
This fails for non-real `n`:
```python
In [1]: import mpmath

In [2]: mpmath.besseli(1j, 1)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 1
----> 1 mpmath.besseli(1j, 1)

File mpmath/mpmath/functions/bessel.py:85, in besseli(ctx, n, z, derivative, **kwargs)
     83 n = ctx.convert(n)
     84 z = ctx.convert(z)
---> 85 if ctx.nint(n) and n < 0:
     86     return ctx.besseli(-n, z, derivative, **kwargs)
     87 if not z:

TypeError: '<' not supported between instances of 'mpc' and 'int'
```

Reply:
Thanks, should be fixed by 294e913.

Reply:
I think it also needs to be changed to `isint` e.g.:
```python
In [2]: mpmath.besseli(-1.5, 1)
Out[2]: mpf('0.2935253263474798')

In [8]: scipy.special.iv(-1.5, 1)
Out[8]: np.float64(-0.2935253263474799)
```